code-studio/src/js/levels/textMatch.js

```diff
@@ -35,8 +35,13 @@ TextMatch.prototype.ready = function () {
     window.getResult = $.proxy(this.getResult, this);
   }
 
-
-  $("#" + this.id + " textarea.response").blur(function () {
+  var textarea = $("#" + this.id + " textarea.response");
+  textarea.blur(function () {
+    if (window.levelGroup && window.levelGroup.answerChangedFn) {
+      window.levelGroup.answerChangedFn(this.levelId, true);
```

Comment:
It looks like this was an existing bug, but one we should fix here and on line 46: We call `this.levelId` inside an unbound callback function.  I suspect `this` refers to the textarea in this case, not the `TextMatch` instance.

This is also a good place for arrow functions and their implicit binding:

```javascript
textarea.blur(() => {
  // Callback
});
textarea.on('input', null, null, () => {
  // Callback
});
```